jenkins/pipelines/java/Jenkinsfile

```diff
@@ -77,45 +72,97 @@ pipeline {
                                 }
                             }
                         }
-                        stage('Windows Web Services Tests') {
-                            when { expression { false } }
+                    }
+                }
+                stage('Linux Tests') {
+                    agent { label 'mob-e2e-deb-02' }
+                    options {
+                        timeout(time: 45, unit: 'MINUTES')
+                        lock("mob-e2e-deb-02")
+                    }
+                    stages {
+                        stage('Linux Desktop Tests') {
                             steps {
-                                pwsh """
-                                    jenkins/pipelines/java/webservice/win_tests.ps1 "${params.CBL_VERSION}" "${params.CBL_BUILD}" "${params.SGW_URL}"
+                                echo "Run Linux Java Desktop Tests"
+                                sh """
+                                    jenkins/pipelines/java/desktop/linux_tests.sh "${params.CBL_VERSION}" "${params.CBL_BUILD}" "${params.SGW_URL}"
                                 """
                             }
                             post {
                                 always {
-                                    echo "Teardown Windows Web Services Tests"
-                                    pwsh '''
-                                        jenkins/pipelines/java/webservice/win_teardown.ps1
+                                    echo "Teardown Linux Java Desktop Tests"
+                                    sh '''
+                                        jenkins/pipelines/java/desktop/linux_teardown.sh
                                     '''
                                 }
                             }
                         }
                     }
                 }
-                stage('Linux Tests') {
-                    agent { label 'mob-e2e-deb-02' }
+            }
+        }
+        stage('Run Webservices Tests') {
+            parallel {
+                stage('OSX Tests') {
+                    agent { label 'mob-e2e-mac-01' }
```

Comment:
In fact these locks won't actually protect against anythiing at the moment since all your jobs that use the same machine run seqentially right?

Reply:
So, in order ;-P
- All of those things that I am locking are lockable resources.  I, too, named them for the devices that host the task
- I'm not running tasks on the same device, in parallel, it is true.  These locks are meant to protect my jobs against YOUR jobs.  Of course that won't work unless you start using the locks, too.  Please do?